### ISSUE TITLE
Heading maker is added to normal text

### ISSUE THREAD

**Issue by fxha** [marktext/marktext#1682]
### Description

The heading maker (`#`) is added when pasting a HTML heading from web page into an existing paragraph (`foo <cursor> bar`).

### Steps to reproduce

1. Create paragraph like `foo <cursor> bar`.
2. Copy HTML heading from web page and paste (`Ctrl+V`) it into preview editor at marked cursor position.

**Expected behavior:**

The heading should be normalized when it's not an empty paragraph because headings in text is not allowed by markdown specification. A heading should be still created if it's allowed, e.g.  empty paragraph or when multiple lines are pasted.

**Actual behavior:**

`foo # My Heading bar`.

### Versions

- Mark Text: v0.15.1 / develop
- Operating system: all

**Comment by Jocs** [comment]
@fxha I pasted and the result is

```md
foo### Descriptionbar
```

Do you mean we need to remove `### `， and only show `fooDescriptionbar`?

**Comment by fxha** [comment]
> @fxha I pasted and the result is
> 
> ```lisp
> foo### Descriptionbar
> ```
> 
> Do you mean we need to remove `### `， and only show `fooDescriptionbar`?

Yes, we should remove the heading marker (here `### `) if we cannot create a heading.

**Comment by Jocs** [comment]
Thank you for the detailed report! This behavior has been fixed in the new `@muyajs/core` editor engine that MarkText now uses. When pasting an HTML heading (or markdown `# Heading`) into a non-empty paragraph, the heading marker (`# `) is now stripped before merging the text inline — so pasting `# My Heading` into `foo <cursor> bar` correctly produces `foo My Headingbar`, not `foo # My Headingbar`. The fix lives in the `inlineMergeText` function (`packages/muya/src/clipboard/paste.ts`) which explicitly drops heading markers when the anchor already has text. Closing as resolved by migration to the new engine.
